tests/test_run_data_managers.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+# Stop pylint complaining from things that are necessary
+# for pytest to work.
+# pylint: disable=no-self-use
+import sys
+import time
+
+import pytest
+import yaml
+from docker_for_galaxy import start_container
+
+from ephemeris import run_data_managers
+from ephemeris.run_data_managers import DataManagers
+from ephemeris.shed_tools import InstallRepositoryManager
+from ephemeris.sleep import galaxy_wait
+
+# Use import to fool flake
+yes_i_am_imported = start_container
+
+
+class TestRunDataManagers(object):
+    """This class tests run-data-managers"""
+
+    def test_install_data_managers(self, start_container):
+        """Install the data_managers on galaxy"""
+        container = start_container
+        data_managers = [
+            dict(name="data_manager_fetch_genome_dbkeys_all_fasta",
+                 owner="devteam"),
+            dict(name="data_manager_sam_fasta_index_builder",
+                 owner="devteam"),
+            dict(name="data_manager_bwa_mem_index_builder",
+                 owner="devteam")
+        ]
+        irm = InstallRepositoryManager(container.gi)
+        irm.install_repositories(data_managers)
+        print("Restart container")
```

Comment:
Is that necessary ? It should work without on the 18.05 image, otherwise that'd be a galaxy bug.

Reply:
I copied this from the old test script (where the container needed to be restarted after new data managers are installed). But I will remove this.

Reply:
```
    def make_post_request(self, url, payload, params=None, files_attached=False):
        """
            Make a POST request using the provided ``url`` and ``payload``.
            The ``payload`` must be a dict that contains the request values.
            The payload dict may contain file handles (in which case the files_attached
            flag must be set to true).
    
            If the ``params`` are not provided, use ``default_params`` class field.
            If params are provided and the provided dict does not have ``key`` key,
            the default ``self.key`` value will be included in what's passed to
            the server via the request.
    
            :return: The decoded response.
            """
        if params is not None and params.get('key', False) is False:
            params['key'] = self.key
        else:
            params = self.default_params
    
        # Compute data, headers, params arguments for request.post,
        # leveraging the requests-toolbelt library if any files have
        # been attached.
        if files_attached:
            payload.update(params)
            payload = MultipartEncoder(fields=payload)
            headers = self.json_headers.copy()
            headers['Content-Type'] = payload.content_type
            post_params = {}
        else:
            payload = json.dumps(payload)
            headers = self.json_headers
            post_params = params
    
        r = requests.post(url, data=payload, headers=headers,
                          verify=self.verify, params=post_params,
                          timeout=self.timeout)
        if r.status_code == 200:
            try:
                return r.json()
            except Exception as e:
                raise ConnectionError("Request was successful, but cannot decode the response content: %s" %
                                      e, body=r.content, status_code=r.status_code)
        # @see self.body for HTTP response body
        raise ConnectionError("Unexpected HTTP status code: %s" % r.status_code,
>                             body=r.text, status_code=r.status_code)
E       bioblend.ConnectionError: Unexpected HTTP status code: 400: {"err_data": {"all_fasta_source": "Parameter all_fasta_source requires a value, but has no legal values defined."}, "err_msg": "Parameter all_fasta_source requires a value, but has no legal values defined.", "err_code": 0}
```

Reply:
Leads to an error unfortunately.
Let's solve this in another PR

Reply:
Yeah, I looked at the old test and I remember why we did this.
Can you add the comment `#We restart galaxy because otherwise the data manager tables won't be watched`